spec/support/event_requiring_a_location.rb

```diff
@@ -0,0 +1,28 @@
+RSpec.shared_examples 'an event requiring a location' do |location_key|
+  it "validates #{location_key} presence" do
+    expect(generic_event).to validate_presence_of(location_key)
+  end
+
+  context "when the #{location_key} does not exist" do
+    let(:named_location) { location_key.to_s.sub('_id', '') }
+
+    before do
+      generic_event.public_send("#{location_key}=", 'flibble')
+    end
+
+    it { is_expected.not_to be_valid }
+
+    it 'attaches a custom message to the generic event' do
```

Comment:
instead of having this context, maybe we can have one test for the id not existing in the database instead:
```
it "validates #{location_key} existing in the database" do
  generic_event.public_send("#{location_key}=", 'flibble')

  expect(generic_event).not_to be_valid
  expect(generic_event.errors[location_key]).to eq(['The location relationship you passed has....'])
end

```

Reply:
On revisiting this with using the `ActiveRecord::RecordNotFound` default error message instead we can test this as follows:
```
  it "validates #{location_key} existing in the database" do
    generic_event.public_send("#{location_key}=", 'flibble')
    
      expect { generic_event.valid? }.to raise_error(ActiveRecord::RecordNotFound, "Couldn't find Location with 'id'=flibble")
  end
```